profiling-ffi/src/profiles.rs

```diff
@@ -691,13 +694,22 @@ pub unsafe extern "C" fn ddog_prof_Profile_serialize(
             ))
         }
     };
+    let old_profile = match profile.reset(start_time.map(SystemTime::from)) {
+        Ok(ok) => ok,
+        Err(err) => {
+            return SerializeResult::Err(Error::from(
+                err.context("ddog_prof_Profile_serialize failed"),
+            ))
+        }
+    };
+
     let end_time = end_time.map(SystemTime::from);
     let duration = match duration_nanos {
         None => None,
         Some(x) if *x < 0 => None,
         Some(x) => Some(Duration::from_nanos((*x) as u64)),
     };
-    match profile.serialize(end_time, duration) {
+    match old_profile.serialize_into_compressed_pprof(end_time, duration) {
         Ok(ok) => SerializeResult::Ok(ok.into()),
         Err(err) => SerializeResult::Err(err.into()),
```

Comment:
Since `serialize_into_compressed_pprof` returns an `anyhow::Result`, I'd convert the error branch to use `err.context("ddog_prof_Profile_serialize failed")` as well.

Reply:
Done